src/mono/mono/metadata/icall.c

```diff
@@ -7243,6 +7243,8 @@ mono_create_icall_signatures (void)
 	}
 }
 
+/* LOCKING: does not take locks. does not use an atomic write to info->wrapper
```

Comment:
Why not?

Reply:
I'll add to the comment.  Because it's called single-threaded early during startup.